programmers/20210514/다트-게임.js

```diff
@@ -0,0 +1,46 @@
+function solution(dartResult) {
+    let temp = dartResult.replace(/[\*+\#+]/g, "");
+    let dartScore = temp.replace(/[A-Z]/g, ",");
+    dartScore = dartScore.split(",");
+    dartScore.pop();
+    const dartBonus = temp.match(/[A-Z]/g);
+    for (let i = 0; i < dartScore.length; i++) {
+        dartScore[i] = Number(dartScore[i]);
+        switch (dartBonus[i]) {
+            case "S": dartBonus[i] = 1; break;
+            case "D": dartBonus[i] = 2; break;
+            case "T": dartBonus[i] = 3; break;
+        }
+    }
+    temp = dartResult.replace(/[0-9]/g, "").replace(/[A-Z]/g, "A");
+    const dartOption = [];
+    let indexOption = -1;
+    for (let i of temp) {
+        if (i === "A") {
+            dartOption.push("")
+            indexOption++;
+        }
+        else dartOption[indexOption] = i;
+    }
+    console.log(dartScore, dartBonus, dartOption);
+    let result = 0;
+    let beforeScore = 0;
+    for (let i = 0; i < dartScore.length; i++) {
+        switch (dartOption[i]) {
+            case "": 
+                result = result + (dartScore[i] ** dartBonus[i]);
+                beforeScore = (dartScore[i] ** dartBonus[i]);
+                break;
+            case "*": 
```

Comment:
Math.pow() 메소드는 제곱메소드 입니다.
```suggestion
    for (let i = 0; i < apart.length; i++) {
        let num = apart[i].match(/\d+/g);
        let char = apart[i].match(/[SDT]/g).join('');
        let specialChar = apart[i].match(/\W/g);
        
        switch (char) {
            case 'S' :
                num = num * 1;
                break;
            case 'D' :
                num = Math.pow(num, 2);
                break;
            case 'T' :
                num = Math.pow(num, 3);
                break;
        }
        
        if (specialChar) {
            if (specialChar.join('') === '*') {
                if(i !== 0) { answer[i - 1] = answer[i - 1] * 2; }
                    num = num * 2;
            } else {
                num = num * -1;
            }
        }
        answer.push(num);
    }
    return answer.reduce((arr1, arr2) => arr1 + arr2);
}
```